src/node.cc

```diff
@@ -1436,7 +1436,21 @@ void FatalException(Isolate* isolate,
       exit(7);
     } else if (caught->IsFalse()) {
       ReportException(env, error, message);
-      exit(1);
+
+      // fatal_exception_function call before may have set a new exit code ->
+      // read it again, otherwise use default for uncaughtException 1
+      Local<String> exit_code = env->exit_code_string();
+      MaybeLocal<Value> maybe_code =
+          process_object->Get(env->context(), exit_code);
+      if (maybe_code.IsEmpty()) {
```

Comment:
This is not really important, but I’m kinda hoping you’ll stick around after your currently open PRs, so as a tip: I personally find it slightly annoying to first have a `MaybeLocal<Something>` and then conditionally converting it to a `Local<Value>`.

What I tend to do then is something along these lines:

```c++
Local<Value> value;
if (!object->Get(context, key).ToLocal(&value) || !value->IsInt32())
  exit(1);
exit(value.As<Int32>()->Value());
```

I know it’s not an exact match to this situation but I hope it’s clear enough what I mean. :)

Reply:
Well, MaybeLocal looks kind of limited. Though I like this neat way of using sce.